node_modules/oae-principals/lib/api.user.js

```diff
@@ -119,63 +126,275 @@ var createUser = module.exports.createUser = function(ctx, tenantAlias, displayN
         }
     }
 
+    var isAdmin = (ctx.user() && ctx.user().isAdmin(tenantAlias));
     opts.visibility = opts.visibility || PrincipalsConfig.getValue(tenantAlias, 'user', 'visibility');
     opts.publicAlias = opts.publicAlias || displayName;
     opts.acceptedTC = opts.acceptedTC || false;
     opts.emailPreference = opts.emailPreference || PrincipalsConfig.getValue(tenantAlias, 'user', 'emailPreference');
+    opts.emailVerified = (opts.emailVerified === true || isAdmin);
 
     var validator = new Validator();
     validator.check(displayName, {'code': 400, 'msg': 'A display name must be provided'}).notEmpty();
     validator.check(displayName, {'code': 400, 'msg': 'A display name can be at most 1000 characters long'}).isShortString();
     validator.check(opts.visibility, {'code': 400, 'msg': 'The specified visibility setting is unknown'}).isIn(_.values(AuthzConstants.visibility));
     validator.check(opts.emailPreference, {'code': 400, 'msg': 'The specified email preference is invalid'}).isIn(_.values(PrincipalsConstants.emailPreferences));
+    if (opts.email) {
+        validator.check(opts.email, {'code': 400, 'msg': 'The specified email address is invalid'}).isEmail();
+    }
 
     if (validator.hasErrors()) {
         return callback(validator.getFirstError());
     }
 
     var id = AuthzUtil.toId('u', tenantAlias, ShortId.generate());
-
-    var values = {
-        'tenantAlias': tenantAlias,
-        'displayName': displayName,
+    var user = new User(tenantAlias, id, displayName, {
         'visibility': opts.visibility,
         'email': opts.email,
-        'emailPreference': opts.emailPreference,
         'locale': opts.locale,
         'publicAlias': opts.publicAlias,
+        'emailPreference': opts.emailPreference,
+        'emailVerified': opts.emailVerified,
         'smallPictureUri': opts.smallPictureUri,
         'mediumPictureUri': opts.mediumPictureUri,
-        'largePictureUri': opts.lagePictureUri
-    };
+        'largePictureUri': opts.lagePictureUri,
+        'acceptedTC': 0,
+        'guest': opts.guest
+    });
 
     // We store the timestamp at which the user accepted the Terms and Conditions
     // This allows for allowing users to re-accept the Terms and Conditions after they have been updated
-    var tcEnabled = PrincipalsConfig.getValue(tenantAlias, 'termsAndConditions', 'enabled');
-    if (tcEnabled && opts.acceptedTC) {
-        // Store the acceptedTC on the return user object
-        opts.acceptedTC = Date.now();
+    user.needsToAcceptTC = PrincipalsConfig.getValue(tenantAlias, 'termsAndConditions', 'enabled');
+    if (user.needsToAcceptTC && opts.acceptedTC) {
```

Comment:
Do we need to stop this when the user needs to accept the TC but hasn't accepted them?